jobs/integration/validation.py

```diff
@@ -2254,15 +2305,27 @@ async def nagios(model, tools):
         app = await model.deploy("nagios", series="bionic")
         await app.expose()
     if "nrpe" not in model.applications:
+        excludes = [
+            "/snap/",
+            "/sys/fs/cgroup",
+            "/run/containerd",
+            "/var/lib/docker",
+            "/run/credentials",
+            "/run/systemd/incoming",
+        ]
         await model.deploy(
             "nrpe",
             series=series,
-            config={"swap": "", "swap_activity": ""},
+            config=dict(
+                swap="", swap_activity="", ro_filesystem_excludes=",".join(excludes)
+            ),
             num_units=0,
             channel="stable",
         )
-    for each in model.applications:
-        if each not in ["nrpe"]:
+    for each, related in model.applications.items():
+        if each not in ["nrpe"] and not any(
+            "nrpe:" in str(rel) for rel in related.relations
+        ):
```

Comment:
if NRPE is already related to the application, skip `re-relating` it to the application.  I found this was necessary while developing the test.  I didn't want to have to keep redeploying nagios and nrpe.  But this was necessary to continue the setup if nrpe was already related to the app